File: Baekjoon/Prefix_Sum/2143.py
# import sys

# input = sys.stdin.readline

# T = int(input())
# n = int(input())
# A = list(map(int, input().split()))
# m = int(input())
# B = list(map(int, input().split()))

# sumA = []
# for s in range(n):
# 	for e in range(s, n):
# 		sumA.append(sum(A[s : e+1]))

# sumB = []
# for s in range(m):
# 	for e in range(s, m):
# 		sumB.append(sum(B[s : e+1]))

# sumB.sort()
# count = 0

# for i in range(len(sumA)):
# 	temp = T - sumA[i]
# 	count += sumB.count(temp)

# print(count)


import sys

input = sys.stdin.readline

T = int(input())
n = int(input())
A = list(map(int, input().split()))
m = int(input())
B = list(map(int, input().split()))

sumA = {}
for s in range(n):
	sum = 0
	for e in range(s, n):
		sum += A[e]
		if sum in sumA:
			sumA[sum] += 1
		else:
			sumA[sum] = 1

count = 0

for s in range(m):
	sum = 0
	for e in range(s, m):
		sum += B[e]
		temp = T - sum
		if temp in sumA:
			count += sumA[temp]

print(count)

File: Baekjoon/Prefix_Sum/2143.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
	int T, n, m;
	cin >> T;

	cin >> n;
	vector<int> A(n);
	for (int i = 0; i < n; i++) {
		cin >> A[i];
	}

	cin >> m;
	vector<int> B(m);
	for (int i = 0; i < m; i++) {
		cin >> B[i];
	}

	vector<int> sumA;
	for (int s = 0; s < n; s++) {
		int sum = 0;
		for (int e = s; e < n; e++) {
			sum += A[e];
			sumA.push_back(sum);
		}
	}

	vector<int> sumB;
	for (int s = 0; s < m; s++) {
		int sum = 0;
		for (int e = s; e < m; e++) {
			sum += B[e];
			sumB.push_back(sum);
		}
	}

	sort(sumB.begin(), sumB.end());
	long long count = 0;

	for (size_t i = 0; i < sumA.size(); i++) {
		int temp = T - sumA[i];
		count += upper_bound(sumB.begin(), sumB.end(), temp) - lower_bound(sumB.begin(), sumB.end(), temp);
	}

	cout << count << endl;

	return 0;
}
